src/Shared/Microsoft.DotNet.Scaffolding.Shared/Project/ProjectModifierHelper.cs

```diff
@@ -41,43 +41,31 @@ internal static async Task<bool> IsMinimalApp(IModelTypesLocator modelTypesLocat
         // Returns true when there is no Startup.cs or equivalent
         internal static async Task<bool> IsMinimalApp(CodeAnalysis.Project project)
         {
-            if (project != null)
+            var startupDocument = project?.Documents.Where(d => d.Name.EndsWith("Startup.cs")).FirstOrDefault() ?? null;
+            if (startupDocument != null)
             {
-                var startupDocument = project.Documents.Where(d => d.Name.EndsWith("Startup.cs")).FirstOrDefault();
-                if (startupDocument == null)
-                {
-                    //if changed the name in Program.cs, get the class name and check.
-                    var programDocument = project.Documents.Where(d => d.Name.EndsWith("Program.cs")).FirstOrDefault();
-                    var startupClassName = await GetStartupClassName(programDocument);
-                    if (!string.IsNullOrEmpty(startupClassName))
-                    {
-                        startupDocument = project.Documents.Where(d => d.Name.EndsWith($"{startupClassName}.cs")).FirstOrDefault();
-                    }
-                }
+                return true;
+            }
 
-                return startupDocument == null;
+            // if changed the name in Program.cs, get the class name and check.
+            var programDocument = project.Documents.Where(d => d.Name.EndsWith("Program.cs")).FirstOrDefault();
```

Comment:
`project` isn't checked for null since the null check was removed.  Change to `project?,Documents` would be good.

Reply:
True, I'm moving the check for null project up earlier so we can catch it and not do any more